src/tools/tools.ts

```diff
@@ -83,59 +84,84 @@ export async function printToolInfo(name: string) {
   }
 }
 
+export function checkToolRequirement(name: string) {
```

Comment:
Oh ok. I thought the switch statement was clever 😅  I was thinking that it would be easier to understand if there was other limitations to add in the futre, if this was organized by tools instead of a bug if else block 😄 Thinking too much ahead probably.

Anyhow, you change looks good. 

Thanks !

Reply:
> I was thinking that it would be easier to understand if there was other limitations to add in the futre, if this was organized by tools instead of a bug if else block

Yes, that's true. But a switch statement + mutable state is a recipe for a bad time in the future, and this refactor gave me an excuse to exercise my [favorite piece of software engineering advice, ever](http://iq0.com/notes/deep.nesting.html).

Reply:
Oh thanks for sharing this article. TBH I had something is mind when choosing `switch()` - I thought (probably too much ahead), that if we had some limitation for each tools, it would be easier to regroup our task by tools and that switch was a nice way to organized. But definitely not the case now, and it should stay simple.

> But a switch statement + mutable state is a recipe for a bad time in the future

Though this was not something I thought about and now I see that tracking `isInstallable` value is definitely harder.

Thanks a lot for taking the time of such feedback ! Really appreciate it! Good to learn new things!